Add AutoEstimator based AutoXGBoost

yushan111 opened the pull request:
This PR includes:

- Add `AutoXGBClassifier` and `AutoXGBRegressor`, which inherit from `AutoEstimator`
- Remove "AutoXGBoost" and "xgb_predictor"
- Add UT for `AutoXGBRegressor` and change interface in AutoXGBoost examples ("AutoXGBoostClassifier" and "AutoXGBoostRegressor")

Interface changes:

- `AutoXGBoost().classifier()` -> `AutoXGBClassifier`, `AutoXGBoost().regressor()` -> `AutoXGBRegressor`
-  `search_alg`, `search_alg_params`, `scheduler`, `scheduler_params` from constructor to fit
-   data format in `fit`: dataframe -> dictionary with keys of ("x", "y", "val_x", "val_y"). Also `feature_cols` and `target_col` which were in constructor are removed. We will improve data format (e.g. split dictionary to "data" and "validation_data") in further PR.

yushan111 commented:
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/5537/

dding3 commented on pyzoo/zoo/orca/automl/xgboost/auto_xgb.py:
```diff
@@ -0,0 +1,52 @@
+class AutoXGBClassifier(AutoEstimator):
+    def __init__(self,
+                 logs_dir="~/auto_xgb_classifier_logs",
+                 n_cpus=None,
```
Can we ask user to set the `n_cpus` in xgb_configs?

dding3 commented on pyzoo/zoo/orca/automl/xgboost/auto_xgb.py:
```diff
@@ -0,0 +1,52 @@
+                 **xgb_configs
+                 ):
+        xgb_model_builder = XGBoostModelBuilder(model_type='classifier',
+                                                n_cpus=n_cpus,
```
Same as above

dding3 reviewed (commented):
The PR overall LGTM. One question is why we ask user to set `n_cpus`, seems like it will be useless if `n_jobs` is also set. Also, we may need add some comments for the apis.

yushan111 commented:
Thanks a lot for the comments! 
Sorry for the confusing `n_cpus`. I have changed it to `cpus_per_trial`. It is similar to `resources_per_trial` and is used for resources scheduling in ray. 
I changed to directly assign the value of `cpus_per_trial` to `n_jobs`. And will raise warning if user pass `n_jobs` a different value from `cpus_per_trial` since the `n_jobs` value will not take any effect.
API doc has been added.

yushan111 commented:
Jenkins: http://10.239.176.111:18888/view/ZOO-PR/job/ZOO-PR-Validation/5557/

Merge first and will open another PR for further comments.
